tests/next_tests/unit_tests/test_constructors.py

```diff
@@ -21,16 +37,49 @@
 sizes = {I: 10, J: 10, K: 10}
 
 
-# TODO: parametrize with gpu backend and compare with cupy array
-@pytest.mark.parametrize(
-    "allocator, device",
-    [
-        [next_allocators.StandardCPUFieldBufferAllocator(), None],
-        [None, core_defs.Device(core_defs.DeviceType.CPU, 0)],
-    ],
-)
-def test_empty(allocator, device):
-    ref = np.empty([sizes[I], sizes[J]]).astype(gtx.float32)
+def _pretty_print(val):
+    if val is None:
+        return "None"
+    if isinstance(val, ModuleType):
+        return val.__name__
+    return val.__class__.__name__
+
+
+def _pretty_print_allocator_device_namespace(val: tuple[Any, Any, Any]):
+    return f"allocator={_pretty_print(val[0])}-device={_pretty_print(val[1])}-ref_namespace={_pretty_print(val[2])}"
+
+
+def allocator_device_refnamespace_params():
+    for v in [
+        [next_allocators.StandardCPUFieldBufferAllocator(), None, np],
+        [None, core_defs.Device(core_defs.DeviceType.CPU, 0), np],
+        [np, None, np],
+    ]:
+        yield pytest.param(
+            v,
+            id=_pretty_print_allocator_device_namespace(v),
+        )
+    for v in [
+        [next_allocators.StandardGPUFieldBufferAllocator(), None, cp],
+        [None, core_defs.Device(core_defs.DeviceType.CUDA, 0), cp],  # TODO CUDA or HIP...
+    ]:
+        yield pytest.param(
+            v, id=_pretty_print_allocator_device_namespace(v), marks=pytest.mark.requires_gpu
+        )
+    for v in [[jnp, None, jnp]]:
+        yield pytest.param(
+            v, id=_pretty_print_allocator_device_namespace(v), marks=pytest.mark.requires_jax
+        )
+
+
+@pytest.fixture(params=allocator_device_refnamespace_params())
+def allocator_device_refnamespace(request):
+    return request.param
+
+
+def test_empty(allocator_device_refnamespace):
```

Comment:
Just a suggestion: we could avoid the intermediate fixture here and just parametrize the test explicitly: 
```suggestion
@pytest.mark.parametrize("allocator, device, xp", [*allocator_device_refnamespace_params()])
def test_empty(allocator, device, xp):
```